api_app/core/serializers.py

```diff
@@ -175,20 +175,19 @@ def _complete_config(
         if plugin_name in visited:
             raise RuntimeError(f"Circular dependency detected in {cls} config")
         visited.add(plugin_name)
+        result = config_dict[plugin_name]
```

Comment:
If you are making this this assignment you are still changing the value of `config_dict[plugin_name]` because for python dict/lists are passed by reference, not value.
Example:
```python3

a = {"key": "value"}
b = a
b["key"] = "value2"
assert a["key"] == "value2"
```